SConstruct

```diff
@@ -205,7 +196,67 @@ if not ('CC' in main and 'CXX' in main):
     error("No C++ compiler installed (package g++ on Ubuntu and RedHat)")
 
 # Find default configuration & binary.
-Default(environ.get('M5_DEFAULT_BINARY', 'build/ARM/gem5.debug'))
+default_target = environ.get('M5_DEFAULT_BINARY', None)
+if default_target:
+    Default(default_target)
+
+# If no target is set, even a default, print help instead.
+if not BUILD_TARGETS:
+    warning("No target specified, and no default.")
+    SetOption('help', True)
+
+buildopts_dir = Dir('#build_opts')
+buildopts = list([f for f in os.listdir(buildopts_dir.abspath) if
+        isfile(os.path.join(buildopts_dir.abspath, f))])
+buildopts.sort()
+
+buildopt_list = '\n'.join(' ' * 10 + buildopt for buildopt in buildopts)
+
+Help(f"""
+Targets:
+        To build gem5 using a predefined configuration, use a target with
+        a directory called "build" in the path, followed by a directory named
+        after a predefined configuration, and then the actual target, likely
```

Comment:
You should specify that these are found in `build_opts`

Reply:
included